### PR TITLE
EdgeObserverLocal, Description for ConnectToEdge Function

### PR DESCRIPTION
**Pull request by ah1053**
**EdgeObserverLocal** for Syncing Local Position, Local Rotation, useful for AR/VR experiences
**WorldOriginTransform** for Redefining the World Origin, if needed
**Description for ConnectToEdge Function**
**NetworkedPlayer** Setting Synced RigidBody to Kinematic (move by script) better than  destroying the Component
**useFallbackLocation** is sat to be true on Non phone devices

### CODE REVIEW

**Review by brucearmstrong — APPROVED** [review]
Looks good.

File: EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            rotationFromServer = transform.localRotation.eulerAngles;
+        }
+
+        private void Update() 
+        {
+            if (EdgeManager.gameStarted)
+            {
+                if ((latestPosition != transform.localPosition && syncOption == SyncOptions.SyncPosition)
+                || (latestRotation != transform.localRotation.eulerAngles && syncOption == SyncOptions.SyncRotation)
+                || ((latestRotation != transform.localRotation.eulerAngles || latestPosition != transform.localPosition) && syncOption == SyncOptions.SyncPositionAndRotation))
```
Comment (franklin-huang-mobiledgex) [inline EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs:69]: These conditionals are a little confusing. Might be useful to have a helper function: bool RequiresUpdate() that either has a switch statement or a series of if elses.

eg.
```
bool RequireUpdate()
{ 
  switch syncOption
  {
    case SyncOptions.SyncPosition:
      if (latestPosition != transform.locationPosition) return true;
    case SyncOptions.SyncRotation:
    case SyncOptions.SyncPositionAndRotation:
  }
}
```

**Review by franklin-huang-mobiledgex — APPROVED** [review]
Looks good! See above for minor comments

File: EdgeMultiplay/Scripts/EdgeMultiplayCallbacks.cs [hunk 2]
```diff
@@ -41,7 +41,12 @@ public class EdgeMultiplayCallbacks : MonoBehaviour
         public static Action gameStart;
         public static Action gameEnd;
 
-        public void ConnectToEdge(bool testingMode = false,bool useFallBackLocation = false)
+        /// <summary>
+        /// Starts the connection to your Edge server, server discovery is based on GPS location and the telecommunication carrier
+        /// </summary>
+        /// <param name="useAnyCarrierNetwork">True by default, set to false to connect to a specific carrier, set carrier name using EdgeManager.integration.carrierName </param>
+        /// <param name="useFallBackLocation">False by default, location is acquired from user GPS location, if you are using location blind device like Oculus, use EdgeManager.integration.SetFallbackLocation()</param>
```
[the remark this answers is not in the document]
Comment (ah1053) [reply EdgeMultiplay/Scripts/EdgeMultiplayCallbacks.cs:48]: Thanks for naming hint

File: EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs (hunk 1 above)
Comment (lgarner) [reply EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs:69]: ++

File: EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        #region EdgeMultiplayObserver Private Functions
+      
+        void ReflectServerData(SyncOptions syncOption)
+        {
+            switch (syncOption)
+            {
+                case SyncOptions.SyncPosition:
+                    if (InterpolatePosition)
+                    {
+                        targetPosInterpolated = Vector3.Lerp(transform.localPosition, positionFromServer, Time.deltaTime * InterpolationFactor);
```
Comment (lgarner) [inline EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs:117]: Lerp is great. Where is InterpolationFactor?

**Review by lgarner — APPROVED** [review]
👍

File: EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs (hunk 3 above)
Comment (ah1053) [reply EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs:138]: EdgeMultiplayObserverLocal inherits from EdgeMultiplayObserver, InterpolationFactor is defined in EdgeMultiplayObserver and sat in the inspector as a float range

File: EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs (hunk 1 above)
Comment (ah1053) [reply EdgeMultiplay/Scripts/EdgeMultiplayObserverLocal.cs]: Done, Many thanks for the review

**Pull request merged**